extensions/virtual-threads/runtime/src/main/java/io/quarkus/virtual/threads/VirtualThreads.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.virtual.threads;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.inject.Qualifier;
+
+/**
+ * Qualifies an injected virtual threads executor service.
+ */
+@Qualifier
+@Target({ FIELD, METHOD, PARAMETER })
+@Retention(RUNTIME)
+public @interface VirtualThreads {
+
+    final class Literal extends AnnotationLiteral<VirtualThreads> implements VirtualThreads {
+
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * Creates a new instance of {@link Literal}.
+         *
+         * @return the literal instance.
+         */
+        public static Literal of() {
```

Comment:
Nit: literals of memberless annotations typically don't have factory methods, but a single field `public static final Literal INSTANCE = new Literal()`.

Reply:
My bad, I knew that! Thanks!